### ISSUE TITLE
Submodules removal cause installation issues

### ISSUE THREAD

**Issue by charlesnguyen8** [BEAMRobotics/beam_robotics#118]
I identified these submodules as having been either deleted or the commit number is no longer valid and causing installation issues.

1. inspection_tools/cnn_defect_segmentation: The GitHub repository is deleted
2. mapping/novatel_span_driver: The GitHub repository is deleted
3. mapping/beam_slam/graph_rviz_plugin: The branch number 59334e4 is not found.

**Comment by nickcharron** [comment]
Hi @thinh241819, the beam_robotics repo was never meant to be a public serving repo. We mostly use this for gathering all projects together in one place for our research lab. That being said, we haven't put much effort into this being easily installed by the public, however, we are happy to have anyone try it out. Let me take a look at these issues and get back to you.

**Comment by nickcharron** [comment]
@thinh241819 out of curiosity, what parts of beam_robotics are you looking to use?

**Comment by nickcharron** [comment]
Looked into these issues:
1. cnn_defect_segmentation wasn't deleted, it was just private (everyone in our lab had access to it though). So I just made it public for you
2. I removed this, we don't need this anymore. I had removed it before, but git is weird and it's very hard to remove a submodule so I didn't remove it in all the right places haha. 
3. Looks like they deleted the branch I was using, so I updated this to main.

It should be all good now if you remove and reclone (with --recursive). I'm re-building everything in beam_robotics now to confirm.

Keep in mind that this has a lot of software dependencies that need to be met, and it's not documented well. Internally we use install script to get all the correct software versions installed. You're welcome to try it, or just look at each of the software componenets it's installing to know what you need to install yourself: https://github.com/BEAMRobotics/beam_install_scripts/blob/master/install.bash

**Comment by nickcharron** [comment]
if beam_slam won't compile, follow the instructions here: https://github.com/BEAMRobotics/beam_slam

**Comment by thinh241819** [comment]
> @thinh241819 out of curiosity, what parts of beam_robotics are you looking to use?

I am actually participating in SRI LAB as an undergrad student. Working on the initial steps of installing this **beam_robotics** repository. 

I am put on a task called **Beam Robotics and LIO-SAM**. Though I am not sure what is the objective of this task, I am just trying to complete the first few steps. Had quite a lot of problems with the **catkin build** step. I tried to go through every sub-folders of the beam_robotics and clone everything by hand. For the deleted repositories, I tried to find their archived version to clone.

Thank you for updating the beam_robotics repo! I will try the installation guide again and let you know the result.

**Comment by thinh241819** [comment]
@nickcharron Thank you so much for your help. I successfully installed everything. So the 3 issues were: 
1. cnn_defect_segmentation and other lost submodules stopped the recursive cloning. 
2. I did not install the **Qwt** library. Installing it with the command `$sudo apt-get install libqwt-qt5-dev` solved most installation issues. Prior to installing Qwt, catkin build would freeze when it was building **libbeam**.
3. Another failed installation is the fuse_optimizers (mapping/beam_slam/dependencies/fuse), which needed me to modify its CMakeLists.txt to add support for OpenMP. 
`
find_package(OpenMP REQUIRED)
if(OpenMP_CXX_FOUND)
    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
endif()
`
Though as you mentioned, this repository is just a collection of modules so I am not sure if **fuse_optimizers** is needed.

I am not familiar with the build system so this took a lot of guessing work. Thank you for your help.
